ocaml/testsuite/tests/typing-layouts-err-msg/concrete.ml

```diff
@@ -0,0 +1,151 @@
+(* TEST
+   flags = "-extension layouts_alpha"
+   * expect
+*)
+
+(**************************)
+(* Concrete layout errors *)
+
+type t_any : any
+type t_void : void
+
+(* Match *)
+let () = match (assert false : t_any) with _ -> ()
+
+[%%expect{|
+type t_any : any
+type t_void : void
+Line 5, characters 15-37:
+5 | let () = match (assert false : t_any) with _ -> ()
+                   ^^^^^^^^^^^^^^^^^^^^^^
+Error: This expression has type t_any but an expression was expected of type
+         ('a : '_representable_layout_1)
+       The layout of t_any is any, because
+         of the definition of t_any at line 1, characters 0-16.
+       But the layout of t_any must be a sublayout of '_representable_layout_1, because
+         it's matched against a pattern.
+|}]
+
+(* Constructor_declaration *)
+type t = A of t_any
+
+[%%expect{|
+Line 1, characters 9-19:
+1 | type t = A of t_any
+             ^^^^^^^^^^
+Error: Constructor argument types must have a representable layout.
+       The layout of t_any is any, because
+         of the definition of t_any at line 1, characters 0-16.
+       But the layout of t_any must be a sublayout of '_representable_layout_2, because
+         it's used as constructor field 0.
+|}]
+
+(* Label_declaration *)
+type t = {a: t_any}
+
+[%%expect{|
+Line 1, characters 10-18:
+1 | type t = {a: t_any}
+              ^^^^^^^^
+Error: Record element types must have a representable layout.
+       The layout of t_any is any, because
+         of the definition of t_any at line 1, characters 0-16.
+       But the layout of t_any must be a sublayout of '_representable_layout_3, because
+         it's used in the declaration of the record field "a/279".
```

Comment:
```suggestion
         it is the type of record field a.
```

Note that I've dropped the quotes and the numerical suffix. (Actually, I think error mesages should do a better job of distinguishing message from quoted code, but that would be a major overhaul.) To get rid of the suffix, you'll have to use a different printer for Idents, I think.